### PR TITLE
WIP - AgentQualifierAuthorizationProvider: fix getAgent() indeterminate beh…

### PR DESCRIPTION
**Pull request by dx9**
______________________________________________ test_bad_agent __________________________________________________________
[gw14] linux2 -- Python 2.7.6 /scratch/cattle.mHqYe/tests/integration/.tox/py27/bin/python2.7
super_client = <cattle.Client object at 0x7fef49bb2190>

```
def test_bad_agent(super_client):
    # Even though super_client will have permissions to create the container
    # event, additional logic should assert that the creator is a valid agent.
    with pytest.raises(ApiError) as e:
        external_id = random_str()
        super_client.create_external_storage_pool_event(
            externalId=external_id,
            eventType="storagepool.create",
            hostUuids=[],
            storagePool={
                'name': 'name-%s' % external_id,
```

> ```
>               'externalId': external_id,
>             })
> ```
> 
> E           Failed: DID NOT RAISE

### CODE REVIEW

File: code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/impl/AgentQualifierAuthorizationProvider.java [hunk 1]
```diff
@@ -121,32 +122,24 @@ protected Long getRawAgentId() {
     }
 
     protected Long getAgent() {
-        ApiRequest request = ApiContext.getContext().getApiRequest();
         Long agentId = getRawAgentId();
         if (agentId != null) {
+            log.error("getAgent(): [{}]", agentId);
```
Comment (ibuildthecloud) [inline code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/impl/AgentQualifierAuthorizationProvider.java]: Why log an error here?

**Comment by ibuildthecloud** [comment]
This changes that behavior slightly in that before is no agents were found it returns null.  Now this will throw an error.  I think if no agents are found it should still return null.

File: code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/impl/AgentQualifierAuthorizationProvider.java (hunk 1 above)
Comment (dx9) [inline code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/impl/AgentQualifierAuthorizationProvider.java]: woops, my bad this is debugging cruft

**Comment by dx9** [comment]
needs more work... introduces the following error in ci

2016-05-13 15:50:14,567 ERROR [:] [] [] [] [ecutorService-4] [o.a.c.m.context.NoExceptionRunnable ] Uncaught exception java.lang.IllegalStateException: java.lang.IllegalStateException: Failed to download [configscripts]
	at io.cattle.platform.agent.connection.simulator.AgentConnectionSimulator.execute(AgentConnectionSimulator.java:46) ~[cattle-simulator-agent-connection-0.5.0-SNAPSHOT.jar:na]
	at io.cattle.platform.agent.server.service.impl.AgentServiceImpl.execute(AgentServiceImpl.java:76) ~[cattle-iaas-agent-server-0.5.0-SNAPSHOT.jar:na]
	at io.cattle.platform.agent.server.eventing.impl.AgentEventListenerImpl.agentRequest(AgentEventListenerImpl.java:15) ~[cattle-iaas-agent-server-0.5.0-SNAPSHOT.jar:na]
	at sun.reflect.GeneratedMethodAccessor900.invoke(Unknown Source) ~[na:na]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:1.7.0_101]
	at java.lang.reflect.Method.invoke(Method.java:606) ~[na:1.7.0_101]
	at io.cattle.platform.eventing.annotation.MethodInvokingListener$1.doWithLockNoResult(MethodInvokingListener.java:76) ~[cattle-framework-eventing-0.5.0-SNAPSHOT.jar:na]
	at io.cattle.platform.lock.LockCallbackNoReturn.doWithLock(LockCallbackNoReturn.java:7) ~[cattle-framework-lock-0.5.0-SNAPSHOT.jar:na]
	at io.cattle.platform.lock.LockCallbackNoReturn.doWithLock(LockCallbackNoReturn.java:3) ~[cattle-framework-lock-0.5.0-SNAPSHOT.jar:na]
	at io.cattle.platform.lock.impl.AbstractLockManagerImpl$3.doWithLock(AbstractLockManagerImpl.java:40) ~[cattle-framework-lock-0.5.0-SNAPSHOT.jar:na]
	at io.cattle.platform.lock.impl.LockManagerImpl.doLock(LockManagerImpl.java:33) ~[cattle-framework-lock-0.5.0-SNAPSHOT.jar:na]
	at io.cattle.platform.lock.impl.AbstractLockManagerImpl.lock(AbstractLockManagerImpl.java:13) ~[cattle-framework-lock-0.5.0-SNAPSHOT.jar:na]
	at io.cattle.platform.lock.impl.AbstractLockManagerImpl.lock(AbstractLockManagerImpl.java:37) ~[cattle-framework-lock-0.5.0-SNAPSHOT.jar:na]
	at io.cattle.platform.eventing.annotation.MethodInvokingListener.onEvent(MethodInvokingListener.java:72) ~[cattle-framework-eventing-0.5.0-SNAPSHOT.jar:na]
	at io.cattle.platform.eventing.impl.AbstractThreadPoolingEventService$2.doRun(AbstractThreadPoolingEventService.java:135) ~[cattle-framework-eventing-0.5.0-SNAPSHOT.jar:na]
	at org.apache.cloudstack.managed.context.NoExceptionRunnable.runInContext(NoExceptionRunnable.java:15) ~[cattle-framework-managed-context-0.5.0-SNAPSHOT.jar:na]
	at org.apache.cloudstack.managed.context.ManagedContextRunnable$1.run(ManagedContextRunnable.java:49) [cattle-framework-managed-context-0.5.0-SNAPSHOT.jar:na]
	at org.apache.cloudstack.managed.context.impl.DefaultManagedContext$1.call(DefaultManagedContext.java:55) [cattle-framework-managed-context-0.5.0-SNAPSHOT.jar:na]
	at org.apache.cloudstack.managed.context.impl.DefaultManagedContext.callWithContext(DefaultManagedContext.java:108) [cattle-framework-managed-context-0.5.0-SNAPSHOT.jar:na]
	at org.apache.cloudstack.managed.context.impl.DefaultManagedContext.runWithContext(DefaultManagedContext.java:52) [cattle-framework-managed-context-0.5.0-SNAPSHOT.jar:na]
	at org.apache.cloudstack.managed.context.ManagedContextRunnable.run(ManagedContextRunnable.java:46) [cattle-framework-managed-context-0.5.0-SNAPSHOT.jar:na]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145) [na:1.7.0_101]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615) [na:1.7.0_101]
	at java.lang.Thread.run(Thread.java:745) [na:1.7.0_101]
Caused by: java.lang.IllegalStateException: Failed to download [configscripts]
	at io.cattle.platform.agent.connection.simulator.impl.SimulatorConfigUpdateProcessor.handle(SimulatorConfigUpdateProcessor.java:59) ~[cattle-simulator-agent-connection-0.5.0-SNAPSHOT.jar:na]
	at io.cattle.platform.agent.connection.simulator.impl.SimulatorConfigUpdateProcessor.handle(SimulatorConfigUpdateProcessor.java:43) ~[cattle-simulator-agent-connection-0.5.0-SNAPSHOT.jar:na]
	at io.cattle.platform.agent.connection.simulator.AgentConnectionSimulator.execute(AgentConnectionSimulator.java:44) ~[cattle-simulator-agent-connection-0.5.0-SNAPSHOT.jar:na]
	... 23 common frames omitted
Caused by: java.io.IOException: Server returned HTTP response code: 422 for URL: http://10.42.0.1:8081/v1/configcontent/configscripts
	at sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1628) ~[na:1.7.0_101]
	at io.cattle.platform.agent.connection.simulator.impl.SimulatorConfigUpdateProcessor.downloadAndPost(SimulatorConfigUpdateProcessor.java:79) ~[cattle-simulator-agent-connection-0.5.0-SNAPSHOT.jar:na]
	at io.cattle.platform.agent.connection.simulator.impl.SimulatorConfigUpdateProcessor.handle(SimulatorConfigUpdateProcessor.java:57) ~[cattle-simulator-agent-connection-0.5.0-SNAPSHOT.jar:na]
	... 25 common frames omitted

**Pull request closed without merging**